From a087b11dd85b856fd7cbae6cd87e60c91e75f80a Mon Sep 17 00:00:00 2001
From: Christophe Giboudeaux <christophe@krop.fr>
Date: Thu, 8 Aug 2019 14:04:40 +0200
Subject: [PATCH] Fix the attica pkgconfig file.

Summary: the file was unusable due to bogus version name, missing dependencies and wrong paths.

Test Plan:
Before:

```
  prefix=/data/kde/inst
  exec_prefix=/data/kde/inst/bin
  libdir=lib64
  includedir=include/KF5Attica

  Name: libKF5Attica
  Description: Qt library to access Open Collaboration Services
  #Requires:
  Version: ..
  Libs: -Llib64 -lKF5Attica
  Cflags: -Iinclude/KF5Attica
```

After:

```
prefix=/data/kde/inst
exec_prefix=${prefix}
libdir=/data/kde/inst/lib64
includedir=/data/kde/inst/include

Name: libKF5Attica
Description: Qt library to access Open Collaboration Services
URL: https://www.kde.org
Requires: Qt5Core Qt5Network
Version: 5.61.0
Libs: -L/data/kde/inst/lib64 -lKF5Attica
Cflags: -I/data/kde/inst/include/KF5/Attica
```

Subscribers: fvogt, kde-frameworks-devel

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D23029
---
 src/CMakeLists.txt              |  1 +
 src/cmake/libKF5Attica.pc.cmake | 19 ++++++++++---------
 src/cmake/libattica.pc.cmake    | 11 -----------
 3 files changed, 11 insertions(+), 20 deletions(-)
 delete mode 100644 src/cmake/libattica.pc.cmake

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4a423b0..27b329b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -199,6 +199,7 @@ install(FILES ${PRI_FILENAME}
 if(NOT WIN32)
     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/libKF5Attica.pc.cmake
         ${CMAKE_CURRENT_BINARY_DIR}/libKF5Attica.pc
+        @ONLY
     )
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libKF5Attica.pc
        DESTINATION ${KDE_INSTALL_LIBDIR}/pkgconfig
diff --git a/src/cmake/libKF5Attica.pc.cmake b/src/cmake/libKF5Attica.pc.cmake
index 75387fa..5dde7c8 100644
--- a/src/cmake/libKF5Attica.pc.cmake
+++ b/src/cmake/libKF5Attica.pc.cmake
@@ -1,11 +1,12 @@
-prefix=${CMAKE_INSTALL_PREFIX}
-exec_prefix=${CMAKE_INSTALL_PREFIX}/bin
-libdir=${LIB_INSTALL_DIR}
-includedir=${INCLUDE_INSTALL_DIR}/KF5Attica
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=@KDE_INSTALL_FULL_LIBDIR@
+includedir=@KDE_INSTALL_FULL_INCLUDEDIR@
 
-Name: lib${ATTICA_LIB_SONAME}
+Name: lib@ATTICA_LIB_SONAME@
 Description: Qt library to access Open Collaboration Services
-#Requires:
-Version: ${CMAKE_LIBATTICA_VERSION_MAJOR}.${CMAKE_LIBATTICA_VERSION_MINOR}.${CMAKE_LIBATTICA_VERSION_PATCH}
-Libs: -L${LIB_INSTALL_DIR} -l${ATTICA_LIB_SONAME}
-Cflags: -I${INCLUDE_INSTALL_DIR}/KF5Attica
+URL: https://www.kde.org
+Requires: Qt5Core Qt5Network
+Version: @ATTICA_VERSION_STRING@
+Libs: -L@KDE_INSTALL_FULL_LIBDIR@ -l@ATTICA_LIB_SONAME@
+Cflags: -I@KDE_INSTALL_FULL_INCLUDEDIR_KF5@/Attica
diff --git a/src/cmake/libattica.pc.cmake b/src/cmake/libattica.pc.cmake
deleted file mode 100644
index 9b73985..0000000
--- a/src/cmake/libattica.pc.cmake
+++ /dev/null
@@ -1,11 +0,0 @@
-prefix=${CMAKE_INSTALL_PREFIX}
-exec_prefix=${CMAKE_INSTALL_PREFIX}/bin
-libdir=${LIB_INSTALL_DIR}
-includedir=${INCLUDE_INSTALL_DIR}
-
-Name: lib${ATTICA_LIB_SONAME}
-Description: Qt library to access Open Collaboration Services
-#Requires:
-Version: ${CMAKE_LIBATTICA_VERSION_MAJOR}.${CMAKE_LIBATTICA_VERSION_MINOR}.${CMAKE_LIBATTICA_VERSION_PATCH}
-Libs: -L${LIB_INSTALL_DIR} -l${ATTICA_LIB_SONAME}
-Cflags: -I${INCLUDE_INSTALL_DIR}
-- 
2.22.0

